rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/20_synonyms_get.yml

```diff
@@ -0,0 +1,129 @@
+setup:
+  - do:
+      synonyms.put:
+        synonyms_set: test-get-synonyms
+        body:
+          synonyms_set:
+            - synonyms: "hello, hi"
+              id: "test-id-1"
+            - synonyms: "bye => goodbye"
+              id: "test-id-2"
+            - synonyms: "test => check"
+              id: "test-id-3"
+
+---
+"Get synonyms set":
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      synonyms.get:
+        synonyms_set: test-get-synonyms
+
+  - match:
+      count: 3
+  - match:
+      synonyms_set:
+        - synonyms: "hello, hi"
+          id: "test-id-1"
+        - synonyms: "bye => goodbye"
+          id: "test-id-2"
+        - synonyms: "test => check"
+          id: "test-id-3"
+
+---
+"Get synonyms set - not found":
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      catch: missing
+      synonyms.get:
+        synonyms_set: unknown-synonym-set
+
+---
+"Pagination - size":
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      synonyms.get:
+        synonyms_set: test-get-synonyms
+        size: 2
+
+  - match:
+      count: 3
+  - match:
+      synonyms_set:
+        - synonyms: "hello, hi"
+          id: "test-id-1"
+        - synonyms: "bye => goodbye"
+          id: "test-id-2"
+
+---
+"Pagination - from":
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      synonyms.get:
+        synonyms_set: test-get-synonyms
+        from: 1
+
+  - match:
+      count: 3
+  - match:
+      synonyms_set:
+        - synonyms: "bye => goodbye"
+          id: "test-id-2"
+        - synonyms: "test => check"
+          id: "test-id-3"
+
+
+---
+"Synonyms set with same IDs":
```

Comment:
This test checks that synonym rules with the same IDs and different synonym sets are retrieved correctly, as Christoph asked for 👍

Reply:
Great test!